Gems/Compression/Code/Source/Tools/CompressionFactoryImpl.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/std/ranges/ranges_algorithm.h>
+#include "CompressionFactoryImpl.h"
+
+namespace Compression
+{
+    CompressionFactoryImpl::~CompressionFactoryImpl()
+    {
+        for (ICompressionInterface* compressionInterface : m_compressionInterfaces)
+        {
+            delete compressionInterface;
+        }
+    }
+    AZStd::span<ICompressionInterface* const> CompressionFactoryImpl::GetCompressionInterfaces()
+    {
+        return m_compressionInterfaces;
+    }
+
+    bool CompressionFactoryImpl::RegisterCompressionInterface(AZStd::unique_ptr<ICompressionInterface>&& compressionInterface)
+    {
+        if (compressionInterface == nullptr)
+        {
+            return false;
+        }
+        auto FindCompression = [compressionAlgorithmId = compressionInterface->GetCompressionAlgorithmId()]
+        (const ICompressionInterface* registeredInterface)
+        {
+            return registeredInterface->GetCompressionAlgorithmId() == compressionAlgorithmId;
+        };
+        if (auto compressionFoundIt = AZStd::ranges::find_if(m_compressionInterfaces, FindCompression);
+            compressionFoundIt != m_compressionInterfaces.end())
+        {
+            return false;
+        }
```

Comment:
The same-ish search code appears several times in this class. Can it be factored into a single function?